packages/core/src/render3/jit/directive.ts

```diff
@@ -186,6 +187,48 @@ export function compileComponent(type: Type<any>, metadata: Component): void {
   });
 }
 
+function getDependencyTypeForError(type: Type<any>) {
+  if (getComponentDef(type)) return 'component';
+  if (getDirectiveDef(type)) return 'directive';
+  if (getPipeDef(type)) return 'pipe';
+  return 'type';
+}
+
+function verifyStandaloneImport(depType: Type<unknown>, importingType: Type<unknown>) {
+  if (isForwardRef(depType)) {
+    depType = resolveForwardRef(depType);
+    if (!depType) {
+      throw new Error(`Expected forwardRef function, imported from "${
+          stringifyForError(importingType)}", to return a standalone entity or NgModule but got "${
+          stringifyForError(depType) || depType}".`);
```

Comment:
```suggestion
          stringifyForError(importingType)}", to return a standalone entity or an NgModule but got "${
          stringifyForError(depType) || depType}".`);
```
Quick question: do we need the `|| depType` here?

Reply:
We don't strictly need it but I did it for cases where something funky (`null` / `undefined` or '') ends up there. So yeh, not necessary but shouldn't do any harm either.

Reply:
False-y values will be handled by the underlying `renderStringify` function, so we should not need it (ok to keep it as well :)).

Reply:
yeh, but `renderStringify` would convert, say `undefined` to an empty string and we wouldn't see it in the error message. I think that it would be preferable to see `undefined` in the error message.

Again, those are minor details.